jms-core/src/main/java/io/micronaut/jms/serdes/DefaultSerializerDeserializer.java

```diff
@@ -138,13 +129,17 @@ public Message serialize(Session session,
                 case BYTES:
                     return serializeBytes(session, (byte[]) body);
                 case OBJECT:
-                    return serializeObject(session, body);
+                    if (body instanceof Serializable) {
+                        return serializeObject(session, (Serializable) body);
+                    } else {
+                        return serializeText(session, objectMapperSupplier.get().writeValueAsString(body));
+                    }
                 case STREAM:
                     return serializeStream(session, (Object[]) body);
                 default:
                     throw new IllegalArgumentException("No known serialization of message " + body);
             }
-        } catch (JMSException | JsonProcessingException | RuntimeException e) {
+        } catch (Throwable e) {
```

Comment:
`Throwable` is too broad, this will catch OOME and other `Error` types that shouldn't be handled here. `Exception` should be enough.

Reply:
or just leave it as it was (along with the other catch change above) since the previous catch clauses were self-documenting of the expected types

Reply:
I'm a bit paranoid about not catching exceptions, this PR fixes one case when the exception was swallowed and left you wondering what is wrong.